spring-ai-integration-tests/src/test/java/org/springframework/ai/integration/tests/tool/ToolCallingManagerTests.java

```diff
@@ -0,0 +1,158 @@
+import org.springframework.ai.chat.model.ChatResponse;
+import org.springframework.ai.chat.prompt.ChatOptions;
+import org.springframework.ai.chat.prompt.Prompt;
+import org.springframework.ai.integration.tests.TestApplication;
+import org.springframework.ai.integration.tests.tool.domain.Author;
+import org.springframework.ai.integration.tests.tool.domain.Book;
+import org.springframework.ai.integration.tests.tool.domain.BookService;
+import org.springframework.ai.model.function.FunctionCallingOptions;
+import org.springframework.ai.model.tool.ToolCallingChatOptions;
+import org.springframework.ai.model.tool.ToolCallingManager;
+import org.springframework.ai.openai.OpenAiChatModel;
+import org.springframework.ai.tool.ToolCallbacks;
+import org.springframework.ai.tool.annotation.Tool;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import reactor.core.publisher.Flux;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Integration tests for {@link ToolCallingManager}.
+ *
+ * @author Thomas Vitale
+ */
+@SpringBootTest(classes = TestApplication.class)
+@EnabledIfEnvironmentVariable(named = "OPENAI_API_KEY", matches = ".*")
+public class ToolCallingManagerTests {
+
+	private final Tools tools = new Tools();
+
+	private final ToolCallingManager toolCallingManager = ToolCallingManager.builder().build();
+
+	@Autowired
+	OpenAiChatModel openAiChatModel;
+
+	@Test
+	void explicitToolCallingExecutionWithNewOptions() {
+		ChatOptions chatOptions = ToolCallingChatOptions.builder()
+			.toolCallbacks(ToolCallbacks.from(tools))
+			.toolExecutionEnabled(false)
+			.build();
+		Prompt prompt = new Prompt(
+				new UserMessage("What books written by %s are available in the library?".formatted("J.R.R. Tolkien")),
+				chatOptions);
+		runExplicitToolCallingExecutionWithOptions(chatOptions, prompt);
+	}
+
+	@Test
+	void explicitToolCallingExecutionWithLegacyOptions() {
+		ChatOptions chatOptions = FunctionCallingOptions.builder()
+			.functionCallbacks(ToolCallbacks.from(tools))
+			.proxyToolCalls(true)
+			.build();
+		Prompt prompt = new Prompt(
+				new UserMessage("What books written by %s are available in the library?".formatted("J.R.R. Tolkien")),
+				chatOptions);
+		runExplicitToolCallingExecutionWithOptions(chatOptions, prompt);
+	}
+
+	@Test
+	void explicitToolCallingExecutionWithNewOptionsStream() {
+		ChatOptions chatOptions = ToolCallingChatOptions.builder()
+			.toolCallbacks(ToolCallbacks.from(tools))
+			.toolExecutionEnabled(false)
+			.build();
+		Prompt prompt = new Prompt(new UserMessage("What books written by %s, %s, and %s are available in the library?"
+			.formatted("J.R.R. Tolkien", "Philip Pullman", "C.S. Lewis")), chatOptions);
+		runExplicitToolCallingExecutionWithOptionsStream(chatOptions, prompt);
+	}
+
+	private void runExplicitToolCallingExecutionWithOptions(ChatOptions chatOptions, Prompt prompt) {
```

Comment:
This shows how to do the tool execution on the client-side. It should be very straightforward using the new ToolCallingManager.